### PR TITLE
removing panic where it should return error

### PR DESCRIPTION
**Pull request by DiegoSena**

### CODE REVIEW

File: internal/inspect/inspect_test.go [hunk 1]
```diff
@@ -39,9 +39,9 @@ func (m *ModelMock) AddInspectInput(payload []byte) int {
 	return args.Int(0)
 }
 
-func (m *ModelMock) GetInspectInput(index int) model.InspectInput {
+func (m *ModelMock) GetInspectInput(index int) (model.InspectInput, error) {
 	args := m.Called(index)
-	return args.Get(0).(model.InspectInput)
+	return args.Get(0).(model.InspectInput), nil
```
Comment (fabiooshiro) [inline internal/inspect/inspect_test.go:44]: The only error I can see is if the args size is zero

File: internal/rollup/rollup_test.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ type SequencerMock struct {
 }
 
 // FinishAndGetNext implements Sequencer.
-func (s *SequencerMock) FinishAndGetNext(accept bool) cModel.Input {
+func (s *SequencerMock) FinishAndGetNext(accept bool) (cModel.Input, error) {
 	panic("unimplemented")
```
Comment (fabiooshiro) [inline internal/rollup/rollup_test.go:48]: we may need to return an error

File: internal/model/state.go [hunk 3]
```diff
@@ -265,11 +290,13 @@ func newRollupsStateInspect(
 	}
 }
 
-func (s *rollupsStateInspect) finish(status cModel.CompletionStatus) {
+func (s *rollupsStateInspect) finish(status cModel.CompletionStatus) error {
 	s.input.Status = status
-	s.input.ProcessedInputCount = s.getProcessedInputCount()
+	inputCount, _ := s.getProcessedInputCount()
```
Comment (fabiooshiro) [inline internal/model/state.go:295]: In this part, we need to verify the error.

File: internal/inspect/inspect_test.go (hunk 1 above)
Comment (fabiooshiro) [reply internal/inspect/inspect_test.go:44]: ow, this is just a mock...

File: internal/model/model.go [hunk 4]
```diff
@@ -222,7 +229,7 @@ func (m *NonodoModel) getProcessedInputCount() int {
 	})
 	total, err := m.inputRepository.Count(ctx, filter)
 	if err != nil {
-		panic(err)
+		return 0, err
```
Comment (fabiooshiro) [inline internal/model/model.go:232]: what about -1?

File: internal/model/model.go [hunk 5]
```diff
@@ -102,14 +103,15 @@ func (m *NonodoModel) AddInspectInput(payload []byte) int {
...
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
 
 	if index >= len(m.inspects) {
-		panic(fmt.Sprintf("invalid inspect input index: %v", index))
+		slog.Error(fmt.Sprintf("invalid inspect input index: %v", index))
+		return InspectInput{}, fmt.Errorf("invalid inspect input index: %v", index)
 	}
-	return *m.inspects[index]
+	return *m.inspects[index], nil
```
[the remark this answers is not in the document]
Comment (DiegoSena) [reply internal/model/model.go:114]: yes, it is being saved in memory. Created issue #151 to work on that.

File: internal/model/state.go [hunk 6]
```diff
@@ -294,7 +321,8 @@ func (s *rollupsStateInspect) addReport(payload []byte) error {
 
 func (s *rollupsStateInspect) registerException(payload []byte) error {
 	s.input.Status = cModel.CompletionStatusException
-	s.input.ProcessedInputCount = s.getProcessedInputCount()
+	inputCount, _ := s.getProcessedInputCount()
```
[the remark this answers is not in the document]
Comment (DiegoSena) [reply internal/model/state.go:331]: done

File: internal/model/state.go (hunk 3 above)
Comment (DiegoSena) [reply internal/model/state.go]: done

File: internal/rollup/rollup_test.go (hunk 2 above)
Comment (DiegoSena) [reply internal/rollup/rollup_test.go]: done

**Review by fabiooshiro — APPROVED** [review]

**Pull request merged**